test/ProgressOnderwijsUtils.Tests/Data/ReadJsonTest.cs

```diff
@@ -0,0 +1,110 @@
+using System.IO.Pipelines;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+namespace ProgressOnderwijsUtils.Tests.Data;
+
+public sealed class ReadJsonTest : TransactedLocalConnection
+{
+    [Fact]
+    public void Utf8JosonWriter_writes_invalid_json_when_aborted()
+    {
+        var pipe = new Pipe();
+        using var writer = new Utf8JsonWriter(pipe.Writer);
+        writer.WriteStartArray();
+        writer.WriteStartObject();
+        writer.WriteString("property", "testje");
+        pipe.Writer.Complete();
+
+        var json = Maybe.Try(() => JsonNode.Parse(Encoding.UTF8.GetString(pipe.Reader.ReadAsync().GetAwaiter().GetResult().Buffer))).Catch<Exception>();
+        PAssert.That(() => json.AssertError().Message.Contains("json", StringComparison.OrdinalIgnoreCase));
+    }
+
+    [Fact]
+    public void Utf8JosonWriter_writes_invalid_json_upon_exception()
+    {
+        var pipe = new Pipe();
+        try {
+            using var writer = new Utf8JsonWriter(pipe.Writer);
+            writer.WriteStartArray();
+            writer.WriteStartObject();
+            writer.WriteString("property", "testje");
+            throw new NotSupportedException();
+        } catch (NotSupportedException) { }
+        pipe.Writer.Complete();
+
+        var json = Maybe.Try(() => JsonNode.Parse(Encoding.UTF8.GetString(pipe.Reader.ReadAsync().GetAwaiter().GetResult().Buffer))).Catch<Exception>();
+        PAssert.That(() => json.AssertError().Message.Contains("json", StringComparison.OrdinalIgnoreCase));
+    }
+
+    [Fact]
+    public void ReadJson_can_read_all_known_used_column_types_from_the_db()
```

Comment:
I think we probably also should have a test that not only approves this json in its string-form, but also re-parses that string form into a poco using System.Text.Json - that way, we can be sure that the choices we make concerning datatype mapping are convenient to parse, too. In essence: have a test that actually consumes this json in at least 1 plausible way we think our clients might.